### ISSUE TITLE
OS X 10.11 crash on startup

### ISSUE THREAD

**Issue by tony** [jonathanslenders/pymux#44]
4c56c8478a1c6142a8cf04e3ecff2f82b78070cb

I haven't ruled out if this is due to 10.11's new features.

See http://stackoverflow.com/a/32906318 (OSError: dlopen(libSystem.dylib, 6): image not found)

```
INFO:pymux:Listening on u'/var/folders/kx/gmc5nl8x2hg370c3nktpc7vm0000gn/T/pymux.sock.me.4'.
INFO:pymux:Client attached.
INFO:pymux:handle command:  <type 'unicode'>.
INFO:pymux:handle command: bind-key '"' split-window -v <type 'unicode'>.
INFO:pymux:handle command: bind-key % split-window -h <type 'unicode'>.
INFO:pymux:handle command: bind-key c new-window <type 'unicode'>.
INFO:pymux:handle command: bind-key Right select-pane -R <type 'unicode'>.
INFO:pymux:handle command: bind-key Left select-pane -L <type 'unicode'>.
INFO:pymux:handle command: bind-key Up select-pane -U <type 'unicode'>.
INFO:pymux:handle command: bind-key Down select-pane -D <type 'unicode'>.
INFO:pymux:handle command: bind-key C-l select-pane -R <type 'unicode'>.
INFO:pymux:handle command: bind-key C-h select-pane -L <type 'unicode'>.
INFO:pymux:handle command: bind-key C-j select-pane -D <type 'unicode'>.
INFO:pymux:handle command: bind-key C-k select-pane -U <type 'unicode'>.
INFO:pymux:handle command: bind-key ; last-pane <type 'unicode'>.
INFO:pymux:handle command: bind-key ! break-pane <type 'unicode'>.
INFO:pymux:handle command: bind-key d detach-client <type 'unicode'>.
INFO:pymux:handle command: bind-key t clock-mode <type 'unicode'>.
INFO:pymux:handle command: bind-key Space next-layout <type 'unicode'>.
INFO:pymux:handle command: bind-key C-z suspend-client <type 'unicode'>.
INFO:pymux:handle command:  <type 'unicode'>.
INFO:pymux:handle command: bind-key z resize-pane -Z <type 'unicode'>.
INFO:pymux:handle command: bind-key k resize-pane -U 2 <type 'unicode'>.
INFO:pymux:handle command: bind-key j resize-pane -D 2 <type 'unicode'>.
INFO:pymux:handle command: bind-key h resize-pane -L 2 <type 'unicode'>.
INFO:pymux:handle command: bind-key l resize-pane -R 2 <type 'unicode'>.
INFO:pymux:handle command: bind-key q display-panes <type 'unicode'>.
INFO:pymux:handle command: bind-key C-Up resize-pane -U 2 <type 'unicode'>.
INFO:pymux:handle command: bind-key C-Down resize-pane -D 2 <type 'unicode'>.
INFO:pymux:handle command: bind-key C-Left resize-pane -L 2 <type 'unicode'>.
INFO:pymux:handle command: bind-key C-Right resize-pane -R 2 <type 'unicode'>.
INFO:pymux:handle command: bind-key M-Up resize-pane -U 5 <type 'unicode'>.
INFO:pymux:handle command: bind-key M-Down resize-pane -D 5 <type 'unicode'>.
INFO:pymux:handle command: bind-key M-Left resize-pane -L 5 <type 'unicode'>.
INFO:pymux:handle command: bind-key M-Right resize-pane -R 5 <type 'unicode'>.
INFO:pymux:handle command:  <type 'unicode'>.
INFO:pymux:handle command: bind-key : command-prompt <type 'unicode'>.
INFO:pymux:handle command: bind-key 0 select-window -t :0 <type 'unicode'>.
INFO:pymux:handle command: bind-key 1 select-window -t :1 <type 'unicode'>.
INFO:pymux:handle command: bind-key 2 select-window -t :2 <type 'unicode'>.
INFO:pymux:handle command: bind-key 3 select-window -t :3 <type 'unicode'>.
INFO:pymux:handle command: bind-key 4 select-window -t :4 <type 'unicode'>.
INFO:pymux:handle command: bind-key 5 select-window -t :5 <type 'unicode'>.
INFO:pymux:handle command: bind-key 6 select-window -t :6 <type 'unicode'>.
INFO:pymux:handle command: bind-key 7 select-window -t :7 <type 'unicode'>.
INFO:pymux:handle command: bind-key 8 select-window -t :8 <type 'unicode'>.
INFO:pymux:handle command: bind-key 9 select-window -t :9 <type 'unicode'>.
INFO:pymux:handle command: bind-key n next-window <type 'unicode'>.
INFO:pymux:handle command: bind-key p previous-window <type 'unicode'>.
INFO:pymux:handle command: bind-key o select-pane -t :.+ <type 'unicode'>.
INFO:pymux:handle command: bind-key { swap-pane -U <type 'unicode'>.
INFO:pymux:handle command: bind-key } swap-pane -D <type 'unicode'>.
INFO:pymux:handle command: bind-key x confirm-before -p "kill-pane #P?" kill-pane <type 'unicode'>.
INFO:pymux:handle command: bind-key & confirm-before -p "kill-window #W?" kill-window <type 'unicode'>.
INFO:pymux:handle command: bind-key C-o rotate-window <type 'unicode'>.
INFO:pymux:handle command: bind-key M-o rotate-window -D <type 'unicode'>.
INFO:pymux:handle command: bind-key C-b send-prefix <type 'unicode'>.
INFO:pymux:handle command: bind-key . command-prompt "move-window -t '%%'" <type 'unicode'>.
INFO:pymux:handle command: bind-key [ copy-mode <type 'unicode'>.
INFO:pymux:handle command: bind-key ] paste-buffer <type 'unicode'>.
INFO:pymux:handle command: bind-key ? list-keys <type 'unicode'>.
INFO:pymux:handle command: bind-key PPage copy-mode -u <type 'unicode'>.
INFO:pymux:handle command:  <type 'unicode'>.
INFO:pymux:handle command: # Layouts. <type 'unicode'>.
INFO:pymux:handle command: bind-key M-1 select-layout even-horizontal <type 'unicode'>.
INFO:pymux:handle command: bind-key M-2 select-layout even-vertical <type 'unicode'>.
INFO:pymux:handle command: bind-key M-3 select-layout main-horizontal <type 'unicode'>.
INFO:pymux:handle command: bind-key M-4 select-layout main-vertical <type 'unicode'>.
INFO:pymux:handle command: bind-key M-5 select-layout tiled <type 'unicode'>.
INFO:pymux:handle command:  <type 'unicode'>.
INFO:pymux:handle command: # Renaming stuff. <type 'unicode'>.
INFO:pymux:handle command: bind-key , command-prompt -I #W "rename-window '%%'" <type 'unicode'>.
INFO:pymux:handle command: #bind-key "'" command-prompt -I #W "rename-pane '%%'" <type 'unicode'>.
INFO:pymux:handle command: bind-key "'" command-prompt -p index "select-window -t ':%%'" <type 'unicode'>.
INFO:pymux:handle command: bind-key . command-prompt "move-window -t '%%'" <type 'unicode'>.
INFO:pymux:Created process ['/bin/zsh'].
```

```
❯ cat /var/folders/kx/gmc5nl8x2hg370c3nktpc7vm0000gn/T/pymux.crash-3X4asm
Traceback (most recent call last):
  File "/Users/me/study/python/pymux/pymux/main.py", line 517, in run_server
    PipeInput(), DummyCallbacks())
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/eventloop/posix.py", line 154, in run
    t()
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/interface.py", line 324, in redraw
    self._redraw()
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/interface.py", line 344, in _redraw
    self.renderer.render(self, self.layout, is_done=self.is_done)
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/renderer.py", line 422, in render
    extended_height=size.rows,
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/containers.py", line 398, in write_to_screen
    self.content.write_to_screen(cli, screen, mouse_handlers, write_position)
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/containers.py", line 142, in write_to_screen
    sizes = self._divide_heigths(cli, write_position)
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/containers.py", line 177, in _divide_heigths
    dimensions = [get_dimension_for_child(c, index) for index, c in enumerate(self.children)]
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/containers.py", line 175, in get_dimension_for_child
    return c.preferred_height(cli, write_position.width, write_position.extended_height)
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/containers.py", line 1614, in preferred_height
    return self.content.preferred_height(cli, width, max_available_height)
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/containers.py", line 254, in preferred_height
    sizes = self._divide_widths(cli, width)
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/containers.py", line 283, in _divide_widths
    dimensions = [get_dimension_for_child(c, index) for index, c in enumerate(self.children)]
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/containers.py", line 281, in get_dimension_for_child
    return c.preferred_width(cli, width)
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/containers.py", line 977, in preferred_width
    cli, max_available_width - total_margin_width)
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/controls.py", line 249, in preferred_width
    text = token_list_to_text(self._get_tokens_cached(cli))
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/controls.py", line 239, in _get_tokens_cached
    cli.render_counter, lambda: self.get_tokens(cli))
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/cache.py", line 35, in get
    value = getter_func()
  File "/Users/me/Library/Python/2.7/lib/python/site-packages/prompt_toolkit/layout/controls.py", line 239, in <lambda>
    cli.render_counter, lambda: self.get_tokens(cli))
  File "/Users/me/study/python/pymux/pymux/layout.py", line 338, in _get_status_tokens
    format_pymux_string(self.pymux, cli, format_str, window=w),
  File "/Users/me/study/python/pymux/pymux/format.py", line 96, in format_pymux_string
    string = string.replace(symbol, f())
  File "/Users/me/study/python/pymux/pymux/format.py", line 47, in name_of_window
    return window.name or '(noname)'
  File "/Users/me/study/python/pymux/pymux/arrangement.py", line 270, in name
    return pane.name
  File "/Users/me/study/python/pymux/pymux/arrangement.py", line 91, in name
    name = self.process.get_name()
  File "/Users/me/study/python/pymux/pymux/process.py", line 336, in get_name
    return get_name_for_fd(self.master)
  File "/Users/me/study/python/pymux/pymux/process.py", line 446, in get_name_for_fd
    return get_proc_name(pgrp)
  File "/Users/me/study/python/pymux/pymux/darwin.py", line 80, in get_proc_name
    proc_kinfo = get_proc_info(pid)
  File "/Users/me/study/python/pymux/pymux/darwin.py", line 54, in get_proc_info
    _init()
  File "/Users/me/study/python/pymux/pymux/darwin.py", line 46, in _init
    LIBC = cdll.LoadLibrary('libc.dylib')
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(libc.dylib, 6): image not found
```

**Comment by jonathanslenders** [comment]
Thanks for pointing out!
This should fix it: https://github.com/jonathanslenders/pymux/commit/bc0e6bad0d9d57a9d14c5abc7be52f83fbdf7e16

Make sure to clone the latest prompt-toolkit before updating pymux. I'll try to push a new release asap.
Jonathan

**Comment by tony** [comment]
Tested and confirmed to fix pymux on OS X 10.11
